### PR TITLE
fix: evm-address metadata leak

### PR DESCRIPTION
**Pull request by daopunk**
## Summary

Stops publishing and reading EVM addresses from public Kind 0 profile metadata, so payout addresses only move through consent-based DM peer-info exchange.

**Before:** Kind 0 could carry `evm_address`, profile sync could fill peer `profiles.evm_address`, and DM send could fall back to that public field when `dm_peer_evm` was missing.

**After:** Kind 0 never publishes or ingests EVM addresses. WalletBar Send/Request unlock only after both sides have a pairwise grant in `dm_peer_evm`. Peer-info requests are consent-only (no `0x` on the wire); accept posts a grant, and the original requester auto-reciprocates so both sides unlock without a second manual step.

## Changes

- Remove Kind 0 EVM publish/ingest and the profile-address fallback for DM send.
- Make `wallet_peer_info_request` address-free; persist peer addresses only from inbound grants.
- Auto-send a reciprocal `wallet_peer_info_grant` after the acceptor’s grant (idempotent via stored `request_id`s).
- Document the private DM wallet peer-info schema and update the manual E2E checklist; retire LF-003.

## Test plan

- [ ] Publish/update profile and confirm Kind 0 metadata has no `evm_address`.
- [ ] In a DM without prior exchange, Send/Request stay locked; “Request wallet information” posts a consent-only request (no `0x`).
- [ ] Peer accepts → grant appears → original requester auto-reciprocates → both sides unlock Send/Request.
- [ ] Peer declines → no `dm_peer_evm` written; either side can request again later.
- [ ] `pnpm test` (especially `wallet-peer-exchange` / `dm-messages` tests) and relevant Rust profile Kind 0 tests pass.

---

[Compound Engineering](https://github.com/EveryInc/compound-engineering-plugin)

### CODE REVIEW

**Comment by logicminds** [comment]
## Code Review Summary

Verdict: **Ready with fixes**.

The core privacy goal is achieved, but the auto-reciprocal grant path and a few consent/validation edges need tightening before merge.

### Key issues to address

| Group | Location | Issue |
|-------|----------|-------|
| Auto-reciprocal grant | `DmThread.svelte:178` | `onSend(grantJson)` is fire-and-forget; state updates and success toast happen even if the send fails. Await the result. |
| Consent enforcement | `DmThread.svelte:133` | Any incoming `wallet_peer_info_grant` has its address saved, even if the user never requested an exchange. Gate persistence on an outbound request or explicit accept. |
| Wallet gating | `WalletBar.svelte:121` | A requester can send a `wallet_peer_info_request` without having a wallet, leaving the acceptor with no reciprocal grant. Preflight and bail early. |
| Reactive state | `DmThread.svelte:116` | The reactive block reads and writes `reciprocatedWalletPeerInfoRequestIds`, which can self-trigger. Derive the should-reciprocate set from a memoized computation over messages. |
| Frontend races | `WalletBar.svelte:94`, `DmThread.svelte:28/30/31` | `npub` can change while async lookups are in flight; capture the active `npub` at the start and ignore stale results. |
| Dead code | `evm_accounts.rs:265`, `profiles.ts:148`, `db.rs:335`, `profile.rs:41` | Remove the dead `#[allow(dead_code)]` helper, dead Kind 0 event listeners, stale comment, and unused `Profile.evm_address` field. |
| Input validation | `dm-messages.ts:47/355`, `+page.svelte:760` | Strengthen npub validation (real bech32), validate `evm_address` in `formatWalletPeerInfoGrant`, and reject accepts for previously-declined requests. |
| Tests | `wallet_ops.rs:374`, `dm-messages.ts`, `wallet-peer-exchange.ts` | Add backend tests that assert `dm_peer_evm`-only resolution and negative/edge tests for the new peer-info parsers and reciprocal helpers. |

### Fix order

1. Await the reciprocal grant send and only update state on success.
2. Gate inbound grant persistence on consent.
3. Require a wallet before sending a peer-info request.
4. Remove dead Kind 0 code.
5. Strengthen input validation.
6. Add privacy-enforcement and reciprocal-flow tests.

Inline callouts are posted on the key lines in the diff.

File: src/components/dm/DmThread.svelte [hunk 1]
```diff
@@ -101,29 +109,84 @@
...
+              alreadyReciprocatedRequestIds: get(reciprocatedWalletPeerInfoRequestIds),
+            })
+          ) {
+            reciprocalGrantInFlight.delete(g.request_id);
+            return;
+          }
+          const grantJson = formatReciprocalWalletPeerGrant({
+            requestId: g.request_id,
+            myNpub: uid,
+            myEvmAddress: myAddr,
```
Comment (logicminds) [inline src/components/dm/DmThread.svelte:194]: **P1** (correctness/security/reliability/frontend-races): This `onSend` is fire-and-forget. Await its result and only update `reciprocatedWalletPeerInfoRequestIds` / show the success toast when it resolves true. See review findings #4–#7.

File: src/components/dm/DmThread.svelte [hunk 2]
```diff
@@ -101,29 +109,84 @@
...
-      if (appliedWalletGrantIds.has(msg.id)) continue;
-      appliedWalletGrantIds.add(msg.id);
-      void setDmPeerEvmAddress(npub, g.evm_address).then(
-        () => {
+
+      if (!appliedWalletGrantIds.has(msg.id)) {
+        appliedWalletGrantIds.add(msg.id);
+        void setDmPeerEvmAddress(npub, g.evm_address).then(
+          () => {
+            dmWalletPeerExchangeTick.update((t: number) => t + 1);
```
Comment (logicminds) [inline src/components/dm/DmThread.svelte:145]: **P1** (adversarial): Inbound grants are persisted here without requiring that the user sent an outbound request first. Gate `setDmPeerEvmAddress` on `threadHasOutboundRequestForId` or an explicit accept to enforce consent. See review findings #3, #9, #26.

File: src/components/wallet/WalletBar.svelte [hunk 3]
```diff
@@ -141,16 +121,9 @@
     }
```
Comment (logicminds) [inline src/components/wallet/WalletBar.svelte:127]: **P2** (correctness/adversarial): `sendWalletInfoRequest` can be sent even if the requester has no EVM address, leaving the acceptor unable to receive a reciprocal grant. Preflight with `getActiveEvmSignerAddress` / `getEvmAddress` and bail early with a toast if neither returns a non-empty address. See review findings #11, #16.

File: src/components/dm/DmThread.svelte [hunk 4]
```diff
@@ -101,29 +109,84 @@
   $: isPactoAppThread = isPactoAppThreadId(npub);
   let appliedWalletGrantIds = new Set<string>();
   let appliedWalletGrantsForNpub: string | null = null;
+  let reciprocalGrantInFlight = new Set<string>();
 
   $: if (npub !== appliedWalletGrantsForNpub) {
     appliedWalletGrantsForNpub = npub;
     appliedWalletGrantIds = new Set();
```
Comment (logicminds) [inline src/components/dm/DmThread.svelte:118]: **P2** (frontend-races): This reactive block both reads and writes `reciprocatedWalletPeerInfoRequestIds`, which can self-trigger. Derive the should-reciprocate set from a memoized computation over messages and only update the persisted store when a confirmed reciprocal is actually sent. See review finding #23.

File: src-tauri/src/evm/evm_accounts.rs [hunk 5]
```diff
@@ -259,21 +259,9 @@ fn new_import_id() -> String {
...
-            }
-        }
-    });
-}
-
-/// Address to publish on Kind 0: `evm_accounts.address` for `default_shared_evm_account_id` (initialized in `ensure_ready`).
+/// Local default receiving address (`default_shared_evm_account_id`); not published on Kind 0.
+/// Kept for DM wallet grants (next steps); not referenced until exchange uses it.
+#[allow(dead_code)]
 pub async fn resolve_default_shared_evm_address_string<R: Runtime>(handle: AppHandle<R>) -> Option<String> {
```
Comment (logicminds) [inline src-tauri/src/evm/evm_accounts.rs]: **P2** (maintainability/project-standards): This helper is marked `#[allow(dead_code)]` and no longer connected to Kind 0 publishing. Remove it now rather than leaving a placeholder that could be re-wired to re-create the public-address leak. See review findings #12, #13.

File: src-tauri/src/profile.rs [hunk 6]
```diff
@@ -37,7 +36,7 @@ pub struct Profile {
     pub avatar_cached: String,
     /// Local cached path for banner image (for offline support)
     pub banner_cached: String,
-    /// Ethereum payout address (0x…): Kind 0 `evm_address` for others; for **mine**, the default-shared wallet account (may differ from active signing address).
+    /// Local-only payout cache (not Kind 0). Cleared on metadata sync; DM pay uses `dm_peer_evm`.
     #[serde(default)]
     pub evm_address: String,
```
Comment (logicminds) [inline src-tauri/src/profile.rs]: **P2** (maintainability/security): `Profile.evm_address` is now always cleared on sync and never used by callers. Remove the field entirely and any DB serialization for it, or add a deprecation note with a follow-up cleanup task. See review findings #21, #22, #40.

File: src/lib/wallet/dm-messages.ts [hunk 7]
```diff
@@ -354,13 +352,12 @@ export function parseWalletPeerInfoRequest(content: string): WalletPeerInfoReque
   if (o.version !== SCHEMA_VERSION) return null;
   if (typeof o.request_id !== 'string' || o.request_id.length === 0) return null;
   if (!isLikelyNpub(o.requester_npub)) return null;
-  if (!isWireEvmAddress(o.requester_evm_address)) return null;
+  // Consent-only: ignore any legacy requester_evm_address on the wire.
```
Comment (logicminds) [inline src/lib/wallet/dm-messages.ts:359]: **P2** (security): `isLikelyNpub` only checks prefix and length. Consider using real bech32 npub validation here. See review finding #32.

File: src-tauri/src/evm/wallet_ops.rs [hunk 8]
```diff
@@ -381,14 +381,11 @@ fn resolve_peer_send_address<R: Runtime>(app: &AppHandle<R>, to_npub: &str) -> R
...
-    let profile_peer = db::get_profile_evm_address(app, to_npub)
-        .map_err(|e| wallet_err_json("DB_ERROR", e, Some(to_npub.to_string())))?;
-    let peer_addr_opt = dm_peer.or(profile_peer);
 
-    let Some(peer_raw) = peer_addr_opt else {
+    let Some(peer_raw) = dm_peer else {
         log::warn!(
             target: "pacto_wallet",
-            "wallet_build_and_send_transaction: missing evm_address for npub prefix={}…",
+            "wallet_build_and_send_transaction: missing dm_peer_evm for npub prefix={}…",
```
Comment (logicminds) [inline src-tauri/src/evm/wallet_ops.rs:385]: **P1** (testing): This path now relies solely on `dm_peer_evm`. Add Rust tests that mock the DB and assert: valid `dm_peer` resolves, missing `dm_peer` returns `MISSING_PEER_EVM_ADDRESS`, and invalid `dm_peer` returns `INVALID_PEER_EVM_ADDRESS`. See review finding #2.

**Comment by logicminds** [comment]
## Callouts for lines not in the diff

A few important findings reference unchanged lines that GitHub does not allow inline review comments on in this PR. Please address them alongside the inline comments above:

- **`src/stores/profiles.ts:148`** — The `kind0_profile_published` / `kind0_profile_publish_failed` listeners are waiting for events that the backend no longer emits after the Kind 0 refactor. Remove the listeners and the corresponding tests.
- **`src/lib/wallet/dm-messages.ts:325`** — `formatWalletPeerInfoGrant` does not validate `evm_address` before serializing. Add an `isWireEvmAddress` check and throw on invalid input.
- **`src-tauri/src/profile.rs:41`** — `Profile.evm_address` is now always cleared and unused. Remove the field and DB serialization, or add a clear deprecation note.
